lib/Runtime/Executor/ThreadPoolExecutor.cpp

```diff
@@ -327,17 +328,19 @@ void ThreadPoolExecutor::propagatePlaceholdersForNode(
   const SymbolTableTy &symbolTable = node->runtimeBundle->getSymbolTable();
 
   for (const auto &symbolPair : symbolTable) {
-    const auto &symbolName = symbolPair.first;
+    const auto &symbolInfo = symbolPair.second;
 
-    auto *placeholder =
-        ctx->getPlaceholderBindings()->getPlaceholderByName(symbolName);
-
-    // If ctx provides a mapping for the symbol, copy it into the context for
-    // the node.
-    if (placeholder) {
-      const auto *tensor = ctx->getPlaceholderBindings()->get(placeholder);
-      executionState->insertIntoNodeCtx(node, symbolName, tensor->clone());
+    if (symbolInfo.symbolCategory != SymbolCategory::Placeholder) {
```

Comment:
we don't need to check `symbolCategory` here, just if `placeholder` is set

Reply:
It should always be `placeholder != nullptr` iff `symbolInfo.symbolCategory != SymbolCategory::Placeholder` so either way works, I can flip which is the condition and which is the assertion but I still want to check that the above is true. Do you think checking the placeholder* is better?

Reply:
I don't think flipping them is any better or worse, if you want to check both here then I think go with whatever you think is more readable. Another alternative is to add a getter for placeholder which does the assert but I don't have strong feelings either way.